### PR TITLE
[libhsplasma/libpcap/lv2/libcrafter] Update to the latest version

### LINKED ISSUE #19880: [libhsplasma] update to 2021.06.08
Library name: libhsplasma

New version number: 2021.06.08

Other information that may be useful (release notes, etc...)
https://github.com/H-uru/libhsplasma/releases/latest

Edit: Fix typo in Library name.

### LINKED ISSUE #19883: [Libpcap] update to 1.10.1
Library name: Libpcap

New version number: 1.10.1

Other information that may be useful (release notes, etc...)
https://www.tcpdump.org/index.html#latest-releases

### LINKED ISSUE #19887: [lv2] update to 1.18.2
Library name: lv2

New version number: 1.18.2

Other information that may be useful (release notes, etc...)
https://lv2plug.in/news.html

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #19880 
  Fixes #19883 
  Fixes #19887

Update `libhsplasma `to 2021.06.08.

Update `libpcap `to 1.10.1, also update repo from GitHub.

Update `lv2 `to 1.18.2, also update repo from GitHub.

Update `libcrafter `to 1.0, also add `supports: ! windows`.

Update ci.baseline.txt.

Note: `Feature libhsplasma[net]` has been passed with the following triplets:

- x86-windows
- x64-windows-static

### CODE REVIEW

File: ports/libcrafter/portfile.cmake [hunk 1]
```diff
@@ -1,25 +1,27 @@
+vcpkg_fail_port_install(ON_TARGET "windows")
```
Comment (NancyLi1013) [inline ports/libcrafter/portfile.cmake:1]: `libcrafter `only supports Unix platform.

File: ports/libpcap/install-pc-on-msvc.patch [hunk 2]
```diff
@@ -1,97 +1,37 @@
...
+         set(LIBS "${LIBS} -l${LIB}")
+     endforeach(LIB)
+-    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/pcap-config.in ${CMAKE_CURRENT_BINARY_DIR}/pcap-config @ONLY)
+     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libpcap.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libpcap.pc @ONLY)
+-    install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/pcap-config DESTINATION bin)
+     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libpcap.pc DESTINATION lib/pkgconfig)
++    
++# On UN*X, and on Windows when not using MSVC, generate libpcap.pc and
++# pcap-config and process man pages and arrange that they be installed.
++if(NOT MSVC)
```
Comment (JackBoosY) [inline ports/libpcap/install-pc-on-msvc.patch:32]: We need always generate the pc file.

File: ports/lv2/vcpkg.json [hunk 3]
```diff
@@ -1,7 +1,17 @@
 {
   "name": "lv2",
-  "version-string": "1.18.0",
+  "version": "1.18.2",
```
Comment (JackBoosY) [inline ports/lv2/vcpkg.json:3]:
```suggestion
  "version-semver": "1.18.2",
```

File: scripts/ci.baseline.txt [hunk 4]
```diff
@@ -671,11 +667,7 @@ libopusenc:arm-uwp=fail
 libopusenc:x64-osx=fail
 libopusenc:x64-uwp=fail
 libosip2:x64-windows-static-md=fail
-libpcap:arm64-windows=fail
-libpcap:arm-uwp=fail
-libpcap:x64-osx=fail
-libpcap:x64-uwp=fail
-libpcap:x64-windows-static=fail
+libpcap:x64-windows-static-md=fail
```
Comment (JackBoosY) [inline scripts/ci.baseline.txt:670]: We should correctly set the CRT linkage instead of adding this.

File: ports/libpcap/portfile.cmake [hunk 5]
```diff
@@ -13,42 +13,29 @@ These can be installed on Ubuntu systems via sudo apt install flex libbison-dev"
...
     PATCHES 
-        0001-fix-package-name.patch
         install-pc-on-msvc.patch
         add-disable-packet-option.patch
 )
 
-# Only dynamic builds are currently supported on Windows
-if(VCPKG_TARGET_IS_WINDOWS)
-    vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
-endif()
```
Comment (JackBoosY) [inline ports/libpcap/portfile.cmake:38]: Does the upstream support static build on Windows now?

File: ports/libhsplasma/portfile.cmake [hunk 6]
```diff
@@ -1,35 +1,32 @@
-vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
```
Comment (JackBoosY) [inline ports/libhsplasma/portfile.cmake:1]: Does the upstream support static build on Windows now?

File: ports/libcrafter/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [inline ports/libcrafter/portfile.cmake:1]: Why failed on Windows now?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libcrafter/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/libcrafter/portfile.cmake:1]: We should patch this line to support static Windows build:
https://github.com/pellegre/libcrafter/blob/master/libcrafter/configure.ac#L39
```
	       -f $withval/lib/libpcap.a; then
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libhsplasma/portfile.cmake (hunk 6 above)
Comment (NancyLi1013) [inline ports/libhsplasma/portfile.cmake:1]: The current version has supported static build, so remove this line.

Comment (NancyLi1013) [reply ports/libhsplasma/portfile.cmake:1]: The current version has supported static build, so remove this line.

File: ports/libpcap/portfile.cmake (hunk 5 above)
Comment (NancyLi1013) [reply ports/libpcap/portfile.cmake:38]: Yes, upstream has supported static build on Windows now.

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the updates!

**Pull request merged**